fda/registration.py

```diff
@@ -0,0 +1,274 @@
+"""Registration of functional data module.
+
+This module contains the methods to perform the registration of
+functional data and related routines, in basis form as well in discretized form.
+
+"""
+
+import numpy
+import scipy.integrate
+
+
+def _check_extrapolation(fd, ext):
+    # Check how to extrapolate
+    # By default uses the value of the fd object
+    if ext == "default" or ext == 0:
+        ext = fd.default_extrapolation
+
+    if ext == "extrapolation" or ext == 1:
+        periodic = 1
+        extrapolation = False
+
+    elif ext == "periodic" or ext == 2:
+        extrapolation = 2
+        periodic = True
+
+    elif ext == "const" or ext == 3:
+        extrapolation = 3
+        periodic = True
+
+    elif ext == "slice" or ext == 4:
+        extrapolation = 1
+        periodic = False
+
+    else:
+        raise ValueError("Incorrect value of ext, should be 'default', 'basis'"
+                         ", 'periodic' or 'slice'.")
+
+    return periodic, extrapolation
+
+
+def shift_registration(fd, maxiter=5, tol=1e-2, ext="default", alpha=1,
+                       initial=[], tfine=[], shifts_array=False, **kwargs):
```

Comment:
It is usually not a good idea in Python to have mutable objects as default values, as they are created only once, at function definition. I think in this case you do not modify the lists, but it is preferable to change them to tuples (immutable) or to None and create the lists in the body.

Reply:
Changed in ffb48cd.